O3 editorial migration guide (#1659)
Financial-Times/origami: components/o-editorial-typography/MIGRATION.md

New typography also supports inverse theme that can be placed on the parent element or on the element itself with `data-o3-theme="inverse"`.

e.g. Core brand with inverse theme will have following markup:

```html
<div data-o3-brand="core" data-o3-theme="inverse">
	<h1 class="o3-editorial-typography__headline">Heading</h1>
</div>
```

e.g Sustainable Views brand with inverse theme specifically applied on the element will have following markup:

```html
<div data-o3-brand="sustainable-views">
	<!-- OTHER HTML -->
	<h1 class="o3-editorial-typography__headline" data-o3-theme="inverse">
		Heading
	</h1>
	<!-- OTHER HTML -->
</div>
```

### Markup

`o3-editorial-typography` includes a JSX template for typography for React users. We recommend using JSX templates instead of copy-pasting HTML markup where possible, though both approaches are supported.

See [Storybook for full o3-editorial-typography JSX documentation](https://o3.origami.ft.com?path=/docs/core-o3-editorial-typography--jsx-documentation). If you choose not to use the `o3-editorial-typography` JSX template, ensure you update your HTML following the [oEditorialTypography mixin guide](#oEditorialTypography).

#### Mixins

> migrating for each mixin remove the mixin from your Sass files. and import `o3-editorial-typography` css instead. For example, replace:
> 
> ```diff
> -@import '@financial-times/o3-editorial-typography/main';
> -@include oEditorialTypographyHeadline;
> 
> +@import '@financial-times/o3-editorial-typography/css/[YOUR BRAND].css';
> ```

Reviewer: It's probably easier to go mixin by mixin, to ensure the correct classes are replaced along then way, than it is to delete all mixins up front. 

```suggestion
To migrate each mixin first import `o3-editorial-typography` css. For example:

```diff
-@import '@financial-times/o3-editorial-typography/main';
+@import '@financial-times/o3-editorial-typography/css/[YOUR BRAND].css';
```
```

#### oEditorialTypography

`oEditorialTypography` mixin was used to include all o-editorial-typography CSS. `o3-editorial-typography` includes all o-editorial-typography CSS by default and can be used by applying correct class names.

> There is no need to include `oEditorialTypography` mixin in your Sass files any more but few things changed in class names.

Reviewer: >but few things changed in class names.

What are these things or how do I figure out what steps to take?

Reviewer: Ahh, I see that's kinda covered below. I don't think that's quite right as `oEditorialTypographyHeadline`, for example, doesn't output a class name. A table to find/replace the classes output from `oEditorialTypography` as for big number would be awesome.

#### oEditorialTypographyHeadline

This mixin outputs large headline text. This style has high visual prominence. This was replaced by a class name `o3-editorial-typography-headline-large`. so:

```html
<h1 class="o-editorial-typography-headline">
	Don't settle for black and white
</h1>
```

becomes:

```html
<h1 class="o3-editorial-typography-headline-large">
	Don't settle for black and white
</h1>
```

> #### oEditorialTypographyHeading

Reviewer: The migration for these mixins aren't quite right as they don't output origami classes. For example the lists mixin is currently used like so

```scss
.my-list {
	@include oEditorialTypographyList();
}

.my-list--ordered {
	@include oEditorialTypographyList('ordered', $include-base-styles: false);
}

.my-list--unordered {
	@include oEditorialTypographyList('unordered', $include-base-styles: false);
}
```

The migration would therefore be to replace custom classes (`my-list` in this case) with an o3 equivalent class.
